### DISCUSSION TITLE
Getting `n` levels of descendants from a commit?

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#3778] [Q&A] [answered]
Is there a way to do the "reverse" of `ancestors(q, n)`? I.e., for a commit `fbd89eaf`, I'd like `fbd89eaf | fbd89eaf+ | fbd89eaf++ | ...`, up to `n` levels.

`descendants` (unlike `ancestors`) doesn't permit a second argument, and `-l n` shows the most recent commits rather than the ones closest to `fbd89eaf`.

**Comment by martinvonz** [comment]
I think the hard part of the work for that is already done and it's just a matter of teaching the `descendants()` function to accept a second argument just like the `ancestors()` function. See the difference here:

https://github.com/martinvonz/jj/blob/a17292317c63df7aad14c3203046f683b847fa69/lib/src/revset.rs#L639-L654

**Reply by BatmanAoD** [reply]
Like this? https://github.com/martinvonz/jj/pull/3780

I haven't tested yet whether that behaves as expected.

**Reply by BatmanAoD** [reply]
...looks like it does, actually!

**Reply by martinvonz** [reply]
Yes, the code looks good to me. Thanks! Maybe you had already found it, but tests seem to be here: https://github.com/martinvonz/jj/blob/e3bb825a21699e704356cb1f262d7cb3be4ed52e/lib/tests/test_revset.rs#L1720-L1819

**Comment by BatmanAoD** [comment, marked as answer]
`descendants(foo, n)` is now supported in `main`, and this will presumably be in 0.19.0.

Without `descendants`, this can be done with `::` and repeated `+`: the equivalent of `descendants(fbd89eaf, 3)` can be written as `fbd89eaf::fbd89eaf++`.
